accord-core/src/main/java/accord/primitives/Ranges.java

```diff
@@ -227,4 +234,17 @@ else if (count == result.length)
         return construct(cachedRanges.completeAndDiscard(result, count));
     }
 
+    public Map<Boolean, Ranges> partitioningBy(Predicate<? super Range> test)
```

Comment:
moved away from the `Partition` class as this actually is better the more I looked at it... `Partition` class didn't properly handle `this.isEmpty()`, but with a map we have a few states

1) map is empty
2) map(true)
3) map(false)
4) map(true, false)

With this change, we at least have the same signature as `Collectors.partitioningBy`